src/uu/ls/src/url_escaper.rs

```diff
@@ -0,0 +1,147 @@
+use std::{borrow::Cow, convert::{TryFrom, TryInto}, ffi::OsStr, ptr, mem::ManuallyDrop};
+
+mod byte_or_escape
+{
+    #[derive(Copy, Clone, PartialEq, Eq)]
+    #[repr(transparent)]
+    pub struct ByteOrEscape(u16);
+
+    impl ByteOrEscape {
+        pub fn byte(b: u8) -> Self {
+            Self(0x8000u16 | (b as u16))
+        }
+
+        pub fn identity(b: u8) -> Self {
+            Self::byte(b)
+        }
+
+        pub fn replacement(b: u8) -> Self {
+            Self::byte(b)
+        }
+
+        pub fn escape(a: u8, b: u8) -> Self {
+            Self((a as u16) | ((b as u16) << 8))
+        }
+
+        #[inline]
+        pub fn classify(self) -> Result<u8, [u8; 2]> {
+            if (self.0 as i16) < 0 {
+                Ok(self.0 as u8)
+            } else {
+                Err([self.0 as u8, (self.0 >> 8) as u8])
+            }
+        }
+
+        pub fn if_byte_1_if_escape_0(self) -> u16 {
```

Comment:
I'm not sure what this function is supposed to do. I can read it, but I cannot match the name to what it's doing.

Reply:
It returns 1 if it's a byte or 0 if it's an escape code. It's summed to count the number of non-escape codes.

Reply:
Ah I see. Maybe it could be called something like `is_byte` and return a `bool`? (Yes, that would require a cast)